include/deal.II/matrix_free/matrix_free.h

```diff
@@ -4435,7 +4435,11 @@ namespace internal
       , operation_before_loop(operation_before_loop)
       , operation_after_loop(operation_after_loop)
       , dof_handler_index_pre_post(dof_handler_index_pre_post)
-    {}
+    {
+      Assert(dst.has_ghost_elements() == false,
+             ExcMessage("The 'dst' vector has read-access to ghost elements."
```

Comment:
Are we using `read-access` somewhere else? This sounds more PETSc language.  What about

```suggestion
             ExcMessage("The destination vector passed to the matrrix-free loop is ghosted."
```